src/ios/actions/leaderScreenActions.js

```diff
@@ -0,0 +1,36 @@
+import * as types from './actionTypes';
+import * as config from '../constants.js';
+
+export function leadersFetchInit() {
+  return {
+    type: types.LEADERS_FETCH_INIT
+  };
+}
+
+export function leadersFetchSuccess(dogs) {
+  return {
+    type: types.LEADERS_FETCH_SUCCESS,
+    dogs: dogs,
+  };
+}
+
+export function leadersFetchFail(error) {
+  return {
+    type: types.LEADERS_FETCH_FAIL,
+    error: error
+  };
+}
+
+export function fetchLeaders() {
+  return function (dispatch) {
+    dispatch(leadersFetchInit())
+      return fetch(config.get_leaders_dogs_url)
+      .then(response => response.json())
+      .then(json => {
+          var dogs = json['dogs'];
+          dispatch(leadersFetchSuccess(dogs))
+      }).catch( (error) => {
+          dispatch(leadersFetchFail(error))
```

Comment:
Lol, real error handling is for people that actually make mistakes.

To any future employers reading my PR history, I swear I'm not normally this cocky. Since you've looked this far into my github, you'll probably see me handle these errors in the future.

But really, I hate QA.